### PR TITLE
move validation type to core

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

This is blocked until fastify-error module is updated (https://github.com/fastify/fastify-error/pull/9)

### CODE REVIEW

File: fastify.d.ts [hunk 1]
```diff
@@ -118,6 +118,20 @@ export type FastifyServerOptions<
...
+  }
+}
+
+export interface ValidationResult {
+  keyword: string;
+  dataPath: string;
+  schemaPath: string;
+  params: Record<string, string | string[]>;
+  message: string;
+}
```
Comment (mcollina) [inline fastify.d.ts:133]: Why is all of this needed?

Comment (Ethan-Arrowood) [reply fastify.d.ts:133]: Not sure. it exists in fastify-error, but the actual validation is done in core so i'm moving the types over to here

**Comment by Ethan-Arrowood** [comment]
fastify-error pr has been merged. When that is released I'll update fastify-error dep here and we can publish this PR for review

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by erfanium** [comment]
@mcollina
When will this change be released? because now, there's no way to use fastify@3.9.2 with fastify-error@0.3.0

**Pull request merged**